sdk/armcore/poller_test.go

```diff
@@ -0,0 +1,157 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package armcore
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/mock"
+)
+
+const (
+	mockSuccessResp = `{"field": "success"}`
+)
+
+type mockType struct {
+	Field string `json:"field,omitempty"`
+}
+
+func getPipeline(srv *mock.Server) azcore.Pipeline {
+	return azcore.NewPipeline(
+		srv,
+		azcore.NewTelemetryPolicy(azcore.TelemetryOptions{}),
+		azcore.NewUniqueRequestIDPolicy(),
+		azcore.NewRetryPolicy(nil),
+		azcore.NewRequestLogPolicy(azcore.RequestLogOptions{}))
+}
+
+func handleError(resp *azcore.Response) error {
+	return fmt.Errorf("error status: %d", resp.StatusCode)
+}
+
+func TestCreatePollerTracker(t *testing.T) {
```

Comment:
The great thing about this is we now have code coverage info for the state machine.  Looks like we need a lot more tests (can do this in a separate PR).

Reply:
Yeah agreed, I was thinking about how far to go with code coverage here vs in the generator which manages to cover many different code paths.